In [1]:
import pandas as pd
import pathlib

In [2]:
df = pd.read_excel("D:/python/NEW1.xlsx")

In [3]:
df

病人编号 性别   年龄  手测大小 性质
0     6900361  女  53岁  13.7  5
1     7826653  男  52岁  11.1  4
2    10554277  男  53岁  10.7  4
3    11086905  女  56岁   8.7  2
4    11086904  女  56岁   6.2  4
..        ... ..  ...   ... ..
437  16306555  女  70岁   NaN  5
438  16313772  男  63岁   NaN  5
439  16331764  女  60岁   NaN  5
440  16357219  女  54岁   NaN  5
441  16405123  女  65岁   NaN  5

[442 rows x 5 columns]

# 训练组非浸润癌列表生成

In [4]:
data_root_1 = pathlib.Path("D:/python/GGO_train/NOIA/")
print(data_root_1)

D:\python\GGO_train\NOIA


In [5]:
all_image_paths_1 = list(item for item in data_root_1.iterdir())
print(all_image_paths_1)

[WindowsPath('D:/python/GGO_train/NOIA/07826653.jpg'), WindowsPath('D:/python/GGO_train/NOIA/10554277.jpg'), WindowsPath('D:/python/GGO_train/NOIA/11086905.jpg'), WindowsPath('D:/python/GGO_train/NOIA/11086906.jpg'), WindowsPath('D:/python/GGO_train/NOIA/11128066.jpg'), WindowsPath('D:/python/GGO_train/NOIA/11219234.jpg'), WindowsPath('D:/python/GGO_train/NOIA/12058330.jpg'), WindowsPath('D:/python/GGO_train/NOIA/12201108.jpg'), WindowsPath('D:/python/GGO_train/NOIA/12930936.jpg'), WindowsPath('D:/python/GGO_train/NOIA/13150501.jpg'), WindowsPath('D:/python/GGO_train/NOIA/13438886.jpg'), WindowsPath('D:/python/GGO_train/NOIA/13621891.jpg'), WindowsPath('D:/python/GGO_train/NOIA/13725093.jpg'), WindowsPath('D:/python/GGO_train/NOIA/13936573.jpg'), WindowsPath('D:/python/GGO_train/NOIA/14206588.jpg'), WindowsPath('D:/python/GGO_train/NOIA/14363830.jpg'), WindowsPath('D:/python/GGO_train/NOIA/14363831.jpg'), WindowsPath('D:/python/GGO_train/NOIA/14363832.jpg'), WindowsPath('D:/python/GGO_

In [6]:
all_image_paths_1 = [str(path) for path in all_image_paths_1]

In [7]:
all_image_paths_1[:74]

['D:\\python\\GGO_train\\NOIA\\07826653.jpg',
 'D:\\python\\GGO_train\\NOIA\\10554277.jpg',
 'D:\\python\\GGO_train\\NOIA\\11086905.jpg',
 'D:\\python\\GGO_train\\NOIA\\11086906.jpg',
 'D:\\python\\GGO_train\\NOIA\\11128066.jpg',
 'D:\\python\\GGO_train\\NOIA\\11219234.jpg',
 'D:\\python\\GGO_train\\NOIA\\12058330.jpg',
 'D:\\python\\GGO_train\\NOIA\\12201108.jpg',
 'D:\\python\\GGO_train\\NOIA\\12930936.jpg',
 'D:\\python\\GGO_train\\NOIA\\13150501.jpg',
 'D:\\python\\GGO_train\\NOIA\\13438886.jpg',
 'D:\\python\\GGO_train\\NOIA\\13621891.jpg',
 'D:\\python\\GGO_train\\NOIA\\13725093.jpg',
 'D:\\python\\GGO_train\\NOIA\\13936573.jpg',
 'D:\\python\\GGO_train\\NOIA\\14206588.jpg',
 'D:\\python\\GGO_train\\NOIA\\14363830.jpg',
 'D:\\python\\GGO_train\\NOIA\\14363831.jpg',
 'D:\\python\\GGO_train\\NOIA\\14363832.jpg',
 'D:\\python\\GGO_train\\NOIA\\14422680.jpg',
 'D:\\python\\GGO_train\\NOIA\\14486667.jpg',
 'D:\\python\\GGO_train\\NOIA\\14603094.jpg',
 'D:\\python\\GGO_train\\NOIA\\146

In [8]:
num = list(item[-12:-4] for item in all_image_paths_1)

In [9]:
num = list(int(item) for item in num)

In [10]:
len(num)

74

In [11]:
num

[7826653,
 10554277,
 11086905,
 11086906,
 11128066,
 11219234,
 12058330,
 12201108,
 12930936,
 13150501,
 13438886,
 13621891,
 13725093,
 13936573,
 14206588,
 14363830,
 14363831,
 14363832,
 14422680,
 14486667,
 14603094,
 14603095,
 14616860,
 14697227,
 14697228,
 14777243,
 14777244,
 14850043,
 14851555,
 14882428,
 14892274,
 14915023,
 14949225,
 14950123,
 15011984,
 15013550,
 15026513,
 15047133,
 15062323,
 15164508,
 15181944,
 15184051,
 15219385,
 15226789,
 15226871,
 15227718,
 15236578,
 15266186,
 15299060,
 15311976,
 15324142,
 15337597,
 15370041,
 15371346,
 15399385,
 15430686,
 15453196,
 15479267,
 15494519,
 15518878,
 15584743,
 15608351,
 15646161,
 15683678,
 15766388,
 15799261,
 15807934,
 15809341,
 15826587,
 15852971,
 15875547,
 15889160,
 15897310,
 15945098]

In [12]:
df1 = pd.DataFrame([])

In [13]:
for i in num:
    c = df.loc[df["病人编号"] == i]
    df1 = df1.append(c,ignore_index=True)

In [14]:
df1

病人编号 性别   年龄  手测大小 性质
0    7826653  男  52岁  11.1  4
1   10554277  男  53岁  10.7  4
2   11086905  女  56岁   8.7  2
3   11086906  女  56岁   6.6  2
4   11128066  男  64岁  15.2  1
..       ... ..  ...   ... ..
69  15852971  女  64岁  16.8  4
70  15875547  女  53岁    14  4
71  15889160  女  29岁  11.9  4
72  15897310  男  66岁   9.9  1
73  15945098  女  52岁    12  3

[74 rows x 5 columns]

In [15]:
df1.loc[:, ["性质"]] = 0

In [16]:
df1

病人编号 性别   年龄  手测大小 性质
0    7826653  男  52岁  11.1  0
1   10554277  男  53岁  10.7  0
2   11086905  女  56岁   8.7  0
3   11086906  女  56岁   6.6  0
4   11128066  男  64岁  15.2  0
..       ... ..  ...   ... ..
69  15852971  女  64岁  16.8  0
70  15875547  女  53岁    14  0
71  15889160  女  29岁  11.9  0
72  15897310  男  66岁   9.9  0
73  15945098  女  52岁    12  0

[74 rows x 5 columns]

# 训练组浸润癌列表生成

In [17]:
data_root_2 = pathlib.Path("D:/python/GGO_train/IA/")
print(data_root_2)

D:\python\GGO_train\IA


In [18]:
all_image_paths_2 = list(item for item in data_root_2.iterdir())
print(all_image_paths_2)

[WindowsPath('D:/python/GGO_train/IA/06900361.jpg'), WindowsPath('D:/python/GGO_train/IA/11124340.jpg'), WindowsPath('D:/python/GGO_train/IA/11126591.jpg'), WindowsPath('D:/python/GGO_train/IA/13386204.jpg'), WindowsPath('D:/python/GGO_train/IA/13959185.jpg'), WindowsPath('D:/python/GGO_train/IA/14301946.jpg'), WindowsPath('D:/python/GGO_train/IA/14431218.jpg'), WindowsPath('D:/python/GGO_train/IA/14548252.jpg'), WindowsPath('D:/python/GGO_train/IA/14614670.jpg'), WindowsPath('D:/python/GGO_train/IA/14665406.jpg'), WindowsPath('D:/python/GGO_train/IA/14667264.jpg'), WindowsPath('D:/python/GGO_train/IA/14683340.jpg'), WindowsPath('D:/python/GGO_train/IA/14755728.jpg'), WindowsPath('D:/python/GGO_train/IA/14762885.jpg'), WindowsPath('D:/python/GGO_train/IA/14768511.jpg'), WindowsPath('D:/python/GGO_train/IA/14828590.jpg'), WindowsPath('D:/python/GGO_train/IA/14851354.jpg'), WindowsPath('D:/python/GGO_train/IA/14854780.jpg'), WindowsPath('D:/python/GGO_train/IA/14866162.jpg'), WindowsPath

In [19]:
all_image_paths_2 = [str(path) for path in all_image_paths_2]

In [20]:
all_image_paths_2[:777]

['D:\\python\\GGO_train\\IA\\06900361.jpg',
 'D:\\python\\GGO_train\\IA\\11124340.jpg',
 'D:\\python\\GGO_train\\IA\\11126591.jpg',
 'D:\\python\\GGO_train\\IA\\13386204.jpg',
 'D:\\python\\GGO_train\\IA\\13959185.jpg',
 'D:\\python\\GGO_train\\IA\\14301946.jpg',
 'D:\\python\\GGO_train\\IA\\14431218.jpg',
 'D:\\python\\GGO_train\\IA\\14548252.jpg',
 'D:\\python\\GGO_train\\IA\\14614670.jpg',
 'D:\\python\\GGO_train\\IA\\14665406.jpg',
 'D:\\python\\GGO_train\\IA\\14667264.jpg',
 'D:\\python\\GGO_train\\IA\\14683340.jpg',
 'D:\\python\\GGO_train\\IA\\14755728.jpg',
 'D:\\python\\GGO_train\\IA\\14762885.jpg',
 'D:\\python\\GGO_train\\IA\\14768511.jpg',
 'D:\\python\\GGO_train\\IA\\14828590.jpg',
 'D:\\python\\GGO_train\\IA\\14851354.jpg',
 'D:\\python\\GGO_train\\IA\\14854780.jpg',
 'D:\\python\\GGO_train\\IA\\14866162.jpg',
 'D:\\python\\GGO_train\\IA\\14867283.jpg',
 'D:\\python\\GGO_train\\IA\\14867501.jpg',
 'D:\\python\\GGO_train\\IA\\14915858.jpg',
 'D:\\python\\GGO_train\\IA\\149

In [21]:
num = list(item[-12:-4] for item in all_image_paths_2)

In [22]:
num = list(int(item) for item in num)

In [23]:
len(num)

77

In [24]:
num

[6900361,
 11124340,
 11126591,
 13386204,
 13959185,
 14301946,
 14431218,
 14548252,
 14614670,
 14665406,
 14667264,
 14683340,
 14755728,
 14762885,
 14768511,
 14828590,
 14851354,
 14854780,
 14866162,
 14867283,
 14867501,
 14915858,
 14934804,
 15043406,
 15064973,
 15066615,
 15070717,
 15075209,
 15093279,
 15125528,
 15140357,
 15149400,
 15163008,
 15164278,
 15187583,
 15211667,
 15218291,
 15225320,
 15250650,
 15259556,
 15261881,
 15286692,
 15337996,
 15363589,
 15395868,
 15399246,
 15410429,
 15421168,
 15433491,
 15448374,
 15507586,
 15552576,
 15604588,
 15609870,
 15625224,
 15632140,
 15642797,
 15654437,
 15666396,
 15673904,
 15715281,
 15721732,
 15723162,
 15727238,
 15737810,
 15774395,
 15779642,
 15827896,
 15828434,
 15850942,
 15859461,
 15864025,
 15867349,
 15869121,
 15882885,
 15945061,
 15945097]

In [25]:
df2 = pd.DataFrame([])

In [26]:
for i in num:
    c = df.loc[df["病人编号"] == i]
    df2 = df2.append(c,ignore_index=True)

In [27]:
df2

病人编号 性别   年龄  手测大小 性质
0    6900361  女  53岁  13.7  5
1   11124340  男  57岁  29.5  5
2   11126591  女  51岁  17.6  5
3   13386204  女  54岁    26  5
4   13959185  女  61岁  17.5  5
..       ... ..  ...   ... ..
72  15867349  女  53岁   8.7  5
73  15869121  女  49岁  14.8  5
74  15882885  女  58岁  29.9  5
75  15945061  女  60岁  17.5  5
76  15945097  男  60岁  18.5  5

[77 rows x 5 columns]

In [28]:
df2.loc[:, ["性质"]] = 1

In [29]:
df2

病人编号 性别   年龄  手测大小 性质
0    6900361  女  53岁  13.7  1
1   11124340  男  57岁  29.5  1
2   11126591  女  51岁  17.6  1
3   13386204  女  54岁    26  1
4   13959185  女  61岁  17.5  1
..       ... ..  ...   ... ..
72  15867349  女  53岁   8.7  1
73  15869121  女  49岁  14.8  1
74  15882885  女  58岁  29.9  1
75  15945061  女  60岁  17.5  1
76  15945097  男  60岁  18.5  1

[77 rows x 5 columns]

# 合并表格

In [43]:
df_train = df1.append(df2, ignore_index=True)

In [44]:
df_train

病人编号 性别   年龄  手测大小 性质
0     7826653  男  52岁  11.1  0
1    10554277  男  53岁  10.7  0
2    11086905  女  56岁   8.7  0
3    11086906  女  56岁   6.6  0
4    11128066  男  64岁  15.2  0
..        ... ..  ...   ... ..
146  15867349  女  53岁   8.7  1
147  15869121  女  49岁  14.8  1
148  15882885  女  58岁  29.9  1
149  15945061  女  60岁  17.5  1
150  15945097  男  60岁  18.5  1

[151 rows x 5 columns]

In [45]:
for i in range(151):
    if df_train.loc[i,"性别"] == "男":
        df_train.loc[i,"性别"] = 1
    else:
        df_train.loc[i,"性别"] = 0

In [46]:
df_train

病人编号 性别   年龄  手测大小 性质
0     7826653  1  52岁  11.1  0
1    10554277  1  53岁  10.7  0
2    11086905  0  56岁   8.7  0
3    11086906  0  56岁   6.6  0
4    11128066  1  64岁  15.2  0
..        ... ..  ...   ... ..
146  15867349  0  53岁   8.7  1
147  15869121  0  49岁  14.8  1
148  15882885  0  58岁  29.9  1
149  15945061  0  60岁  17.5  1
150  15945097  1  60岁  18.5  1

[151 rows x 5 columns]

In [47]:
for i in range(151):
    age = df_train.loc[i,"年龄"]
    age = age[0:2]
    df_train.loc[i,"年龄"] = age

In [48]:
df_train

病人编号 性别  年龄  手测大小 性质
0     7826653  1  52  11.1  0
1    10554277  1  53  10.7  0
2    11086905  0  56   8.7  0
3    11086906  0  56   6.6  0
4    11128066  1  64  15.2  0
..        ... ..  ..   ... ..
146  15867349  0  53   8.7  1
147  15869121  0  49  14.8  1
148  15882885  0  58  29.9  1
149  15945061  0  60  17.5  1
150  15945097  1  60  18.5  1

[151 rows x 5 columns]

In [49]:
df_train.columns=["patient_num","sex","age","diameter","type"]  

In [50]:
df_train

patient_num sex age diameter type
0       7826653   1  52     11.1    0
1      10554277   1  53     10.7    0
2      11086905   0  56      8.7    0
3      11086906   0  56      6.6    0
4      11128066   1  64     15.2    0
..          ...  ..  ..      ...  ...
146    15867349   0  53      8.7    1
147    15869121   0  49     14.8    1
148    15882885   0  58     29.9    1
149    15945061   0  60     17.5    1
150    15945097   1  60     18.5    1

[151 rows x 5 columns]

In [51]:
train_group = [0]*151

In [52]:
df_train.insert(loc=0, column='group', value=train_group)

In [53]:
pre_train = np.load("D:/python/pre_train.npy")

In [54]:
df_train.insert(loc=5, column='deepscore', value=pre_train)

In [55]:
df_train

group patient_num sex age diameter  deepscore type
0        0     7826653   1  52     11.1   0.965127    0
1        0    10554277   1  53     10.7   0.515213    0
2        0    11086905   0  56      8.7   0.876166    0
3        0    11086906   0  56      6.6   0.582331    0
4        0    11128066   1  64     15.2   0.791398    0
..     ...         ...  ..  ..      ...        ...  ...
146      0    15867349   0  53      8.7   0.589639    1
147      0    15869121   0  49     14.8   0.115247    1
148      0    15882885   0  58     29.9   0.051394    1
149      0    15945061   0  60     17.5   0.125684    1
150      0    15945097   1  60     18.5   0.119397    1

[151 rows x 7 columns]

In [56]:
df_train.to_csv('D:/python/df_train.csv')

# 验证组非浸润癌列表生成

In [57]:
data_root_3 = pathlib.Path("D:/python/GGO_test/NOIA/")
print(data_root_3)

D:\python\GGO_test\NOIA


In [58]:
all_image_paths_3 = list(item for item in data_root_3.iterdir())
print(all_image_paths_3)

[WindowsPath('D:/python/GGO_test/NOIA/15948741.jpg'), WindowsPath('D:/python/GGO_test/NOIA/15953722.jpg'), WindowsPath('D:/python/GGO_test/NOIA/15969517.jpg'), WindowsPath('D:/python/GGO_test/NOIA/15973818.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16002097.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16013798.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16015123.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16016891.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16037928.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16037929.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16061526.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16115683.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16125690.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16138538.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16146671.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16179449.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16190391.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16205881.jpg'), WindowsPath('D:/python/GGO_test/NOIA/16246061

In [59]:
all_image_paths_3 = [str(path) for path in all_image_paths_3]

In [60]:
all_image_paths_1[:30]

['D:\\python\\GGO_train\\NOIA\\07826653.jpg',
 'D:\\python\\GGO_train\\NOIA\\10554277.jpg',
 'D:\\python\\GGO_train\\NOIA\\11086905.jpg',
 'D:\\python\\GGO_train\\NOIA\\11086906.jpg',
 'D:\\python\\GGO_train\\NOIA\\11128066.jpg',
 'D:\\python\\GGO_train\\NOIA\\11219234.jpg',
 'D:\\python\\GGO_train\\NOIA\\12058330.jpg',
 'D:\\python\\GGO_train\\NOIA\\12201108.jpg',
 'D:\\python\\GGO_train\\NOIA\\12930936.jpg',
 'D:\\python\\GGO_train\\NOIA\\13150501.jpg',
 'D:\\python\\GGO_train\\NOIA\\13438886.jpg',
 'D:\\python\\GGO_train\\NOIA\\13621891.jpg',
 'D:\\python\\GGO_train\\NOIA\\13725093.jpg',
 'D:\\python\\GGO_train\\NOIA\\13936573.jpg',
 'D:\\python\\GGO_train\\NOIA\\14206588.jpg',
 'D:\\python\\GGO_train\\NOIA\\14363830.jpg',
 'D:\\python\\GGO_train\\NOIA\\14363831.jpg',
 'D:\\python\\GGO_train\\NOIA\\14363832.jpg',
 'D:\\python\\GGO_train\\NOIA\\14422680.jpg',
 'D:\\python\\GGO_train\\NOIA\\14486667.jpg',
 'D:\\python\\GGO_train\\NOIA\\14603094.jpg',
 'D:\\python\\GGO_train\\NOIA\\146

In [61]:
num = list(item[-12:-4] for item in all_image_paths_3)

In [62]:
num = list(int(item) for item in num)

In [63]:
len(num)

30

In [64]:
num

[15948741,
 15953722,
 15969517,
 15973818,
 16002097,
 16013798,
 16015123,
 16016891,
 16037928,
 16037929,
 16061526,
 16115683,
 16125690,
 16138538,
 16146671,
 16179449,
 16190391,
 16205881,
 16246061,
 16249434,
 16292540,
 16293872,
 16327554,
 16332422,
 16341295,
 16344725,
 16356575,
 16372828,
 16380887,
 16388544]

In [65]:
df3 = pd.DataFrame([])

In [66]:
for i in num:
    c = df.loc[df["病人编号"] == i]
    df3 = df3.append(c,ignore_index=True)

In [67]:
df3

病人编号 性别   年龄  手测大小 性质
0   15948741  女  49岁    13  2
1   15953722  男  49岁  14.3  4
2   15969517  女  67岁    11  4
3   15973818  女  36岁   8.6  4
4   16002097  女  51岁  10.1  4
5   16013798  女  64岁  15.8  4
6   16015123  男  62岁  21.4  2
7   16016891  女  54岁  11.1  4
8   16037928  女  42岁  15.8  1
9   16037929  女  42岁  11.3  1
10  16061526  女  52岁   6.9  4
11  16115683  女  52岁  15.5  4
12  16125690  女  47岁  2.91  4
13  16138538  男  61岁  11.2  1
14  16146671  女  59岁  13.2  4
15  16179449  女  65岁  22.4  4
16  16190391  女  61岁  12.1  4
17  16205881  女  55岁   9.6  1
18  16246061  女  63岁  12.9  4
19  16249434  女  55岁  17.3  2
20  16292540  女  64岁  16.7  4
21  16293872  女  39岁  11.6  2
22  16327554  女  55岁   6.3  1
23  16332422  男  58岁  13.3  4
24  16341295  女  58岁   9.9  4
25  16344725  女  64岁  13.1  4
26  16356575  女  58岁  14.1  2
27  16372828  女  57岁  17.6  1
28  16380887  女  55岁   8.8  4
29  16388544  女  61岁  11.5  3

In [68]:
df3.loc[:, ["性质"]] = 0

In [69]:
df3

病人编号 性别   年龄  手测大小 性质
0   15948741  女  49岁    13  0
1   15953722  男  49岁  14.3  0
2   15969517  女  67岁    11  0
3   15973818  女  36岁   8.6  0
4   16002097  女  51岁  10.1  0
5   16013798  女  64岁  15.8  0
6   16015123  男  62岁  21.4  0
7   16016891  女  54岁  11.1  0
8   16037928  女  42岁  15.8  0
9   16037929  女  42岁  11.3  0
10  16061526  女  52岁   6.9  0
11  16115683  女  52岁  15.5  0
12  16125690  女  47岁  2.91  0
13  16138538  男  61岁  11.2  0
14  16146671  女  59岁  13.2  0
15  16179449  女  65岁  22.4  0
16  16190391  女  61岁  12.1  0
17  16205881  女  55岁   9.6  0
18  16246061  女  63岁  12.9  0
19  16249434  女  55岁  17.3  0
20  16292540  女  64岁  16.7  0
21  16293872  女  39岁  11.6  0
22  16327554  女  55岁   6.3  0
23  16332422  男  58岁  13.3  0
24  16341295  女  58岁   9.9  0
25  16344725  女  64岁  13.1  0
26  16356575  女  58岁  14.1  0
27  16372828  女  57岁  17.6  0
28  16380887  女  55岁   8.8  0
29  16388544  女  61岁  11.5  0

# 训练组浸润癌列表生成

In [70]:
data_root_4 = pathlib.Path("D:/python/GGO_test/IA/")
print(data_root_4)

D:\python\GGO_test\IA


In [71]:
all_image_paths_4 = list(item for item in data_root_4.iterdir())
print(all_image_paths_4)

[WindowsPath('D:/python/GGO_test/IA/15957645.jpg'), WindowsPath('D:/python/GGO_test/IA/15959386.jpg'), WindowsPath('D:/python/GGO_test/IA/15975538.jpg'), WindowsPath('D:/python/GGO_test/IA/15987050.jpg'), WindowsPath('D:/python/GGO_test/IA/15999229.jpg'), WindowsPath('D:/python/GGO_test/IA/16030518.jpg'), WindowsPath('D:/python/GGO_test/IA/16031284.jpg'), WindowsPath('D:/python/GGO_test/IA/16031822.jpg'), WindowsPath('D:/python/GGO_test/IA/16037970.jpg'), WindowsPath('D:/python/GGO_test/IA/16059736.jpg'), WindowsPath('D:/python/GGO_test/IA/16065569.jpg'), WindowsPath('D:/python/GGO_test/IA/16086039.jpg'), WindowsPath('D:/python/GGO_test/IA/16108185.jpg'), WindowsPath('D:/python/GGO_test/IA/16147283.jpg'), WindowsPath('D:/python/GGO_test/IA/16159443.jpg'), WindowsPath('D:/python/GGO_test/IA/16171595.jpg'), WindowsPath('D:/python/GGO_test/IA/16182203.jpg'), WindowsPath('D:/python/GGO_test/IA/16203411.jpg'), WindowsPath('D:/python/GGO_test/IA/16208480.jpg'), WindowsPath('D:/python/GGO_tes

In [72]:
all_image_paths_4 = [str(path) for path in all_image_paths_4]

In [73]:
all_image_paths_2[:29]

['D:\\python\\GGO_train\\IA\\06900361.jpg',
 'D:\\python\\GGO_train\\IA\\11124340.jpg',
 'D:\\python\\GGO_train\\IA\\11126591.jpg',
 'D:\\python\\GGO_train\\IA\\13386204.jpg',
 'D:\\python\\GGO_train\\IA\\13959185.jpg',
 'D:\\python\\GGO_train\\IA\\14301946.jpg',
 'D:\\python\\GGO_train\\IA\\14431218.jpg',
 'D:\\python\\GGO_train\\IA\\14548252.jpg',
 'D:\\python\\GGO_train\\IA\\14614670.jpg',
 'D:\\python\\GGO_train\\IA\\14665406.jpg',
 'D:\\python\\GGO_train\\IA\\14667264.jpg',
 'D:\\python\\GGO_train\\IA\\14683340.jpg',
 'D:\\python\\GGO_train\\IA\\14755728.jpg',
 'D:\\python\\GGO_train\\IA\\14762885.jpg',
 'D:\\python\\GGO_train\\IA\\14768511.jpg',
 'D:\\python\\GGO_train\\IA\\14828590.jpg',
 'D:\\python\\GGO_train\\IA\\14851354.jpg',
 'D:\\python\\GGO_train\\IA\\14854780.jpg',
 'D:\\python\\GGO_train\\IA\\14866162.jpg',
 'D:\\python\\GGO_train\\IA\\14867283.jpg',
 'D:\\python\\GGO_train\\IA\\14867501.jpg',
 'D:\\python\\GGO_train\\IA\\14915858.jpg',
 'D:\\python\\GGO_train\\IA\\149

In [74]:
num = list(item[-12:-4] for item in all_image_paths_4)

In [75]:
num = list(int(item) for item in num)

In [76]:
len(num)

29

In [77]:
num

[15957645,
 15959386,
 15975538,
 15987050,
 15999229,
 16030518,
 16031284,
 16031822,
 16037970,
 16059736,
 16065569,
 16086039,
 16108185,
 16147283,
 16159443,
 16171595,
 16182203,
 16203411,
 16208480,
 16228130,
 16238261,
 16246609,
 16249435,
 16253143,
 16293130,
 16335597,
 16351149,
 16360047,
 16393326]

In [78]:
df4 = pd.DataFrame([])

In [79]:
for i in num:
    c = df.loc[df["病人编号"] == i]
    df4 = df4.append(c,ignore_index=True)

In [80]:
df4

病人编号 性别   年龄  手测大小 性质
0   15957645  女  54岁    25  5
1   15959386  女  57岁  23.6  5
2   15975538  男  56岁  20.5  5
3   15987050  女  49岁  13.5  5
4   15999229  男  59岁  2.62  5
5   16030518  女  59岁  17.2  5
6   16031284  男  52岁  21.1  5
7   16031822  女  58岁  19.3  5
8   16037970  男  46岁    30  5
9   16059736  女  53岁  15.3  5
10  16065569  男  54岁  17.8  5
11  16086039  女  57岁  18.9  5
12  16108185  女  29岁  16.8  5
13  16147283  女  60岁  25.1  5
14  16159443  男  60岁  12.7  5
15  16171595  女  52岁    17  5
16  16182203  男  54岁  16.6  5
17  16203411  女  57岁  14.7  5
18  16208480  女  55岁  21.5  5
19  16228130  女  59岁  16.5  5
20  16238261  男  49岁  24.6  5
21  16246609  女  60岁  26.4  5
22  16249435  女  55岁   8.5  2
23  16253143  男  52岁  22.9  5
24  16293130  女  59岁  28.7  5
25  16335597  男  60岁  18.8  5
26  16351149  女  56岁  22.2  5
27  16360047  女  57岁  15.6  5
28  16393326  女  51岁  18.1  5

In [81]:
df4.loc[:, ["性质"]] = 1

In [82]:
df4

病人编号 性别   年龄  手测大小 性质
0   15957645  女  54岁    25  1
1   15959386  女  57岁  23.6  1
2   15975538  男  56岁  20.5  1
3   15987050  女  49岁  13.5  1
4   15999229  男  59岁  2.62  1
5   16030518  女  59岁  17.2  1
6   16031284  男  52岁  21.1  1
7   16031822  女  58岁  19.3  1
8   16037970  男  46岁    30  1
9   16059736  女  53岁  15.3  1
10  16065569  男  54岁  17.8  1
11  16086039  女  57岁  18.9  1
12  16108185  女  29岁  16.8  1
13  16147283  女  60岁  25.1  1
14  16159443  男  60岁  12.7  1
15  16171595  女  52岁    17  1
16  16182203  男  54岁  16.6  1
17  16203411  女  57岁  14.7  1
18  16208480  女  55岁  21.5  1
19  16228130  女  59岁  16.5  1
20  16238261  男  49岁  24.6  1
21  16246609  女  60岁  26.4  1
22  16249435  女  55岁   8.5  1
23  16253143  男  52岁  22.9  1
24  16293130  女  59岁  28.7  1
25  16335597  男  60岁  18.8  1
26  16351149  女  56岁  22.2  1
27  16360047  女  57岁  15.6  1
28  16393326  女  51岁  18.1  1

# 合并表格

In [83]:
df_test = df3.append(df4, ignore_index=True)

In [84]:
df_test

病人编号 性别   年龄  手测大小 性质
0   15948741  女  49岁    13  0
1   15953722  男  49岁  14.3  0
2   15969517  女  67岁    11  0
3   15973818  女  36岁   8.6  0
4   16002097  女  51岁  10.1  0
5   16013798  女  64岁  15.8  0
6   16015123  男  62岁  21.4  0
7   16016891  女  54岁  11.1  0
8   16037928  女  42岁  15.8  0
9   16037929  女  42岁  11.3  0
10  16061526  女  52岁   6.9  0
11  16115683  女  52岁  15.5  0
12  16125690  女  47岁  2.91  0
13  16138538  男  61岁  11.2  0
14  16146671  女  59岁  13.2  0
15  16179449  女  65岁  22.4  0
16  16190391  女  61岁  12.1  0
17  16205881  女  55岁   9.6  0
18  16246061  女  63岁  12.9  0
19  16249434  女  55岁  17.3  0
20  16292540  女  64岁  16.7  0
21  16293872  女  39岁  11.6  0
22  16327554  女  55岁   6.3  0
23  16332422  男  58岁  13.3  0
24  16341295  女  58岁   9.9  0
25  16344725  女  64岁  13.1  0
26  16356575  女  58岁  14.1  0
27  16372828  女  57岁  17.6  0
28  16380887  女  55岁   8.8  0
29  16388544  女  61岁  11.5  0
30  15957645  女  54岁    25  1
31  15959386  女  57岁  23.6  1
32  15975538  男  56岁  20.5  1
33  15987050  女  49岁  13.5  1
34  15999229  男  59岁  2.62  1
35  16030518  女  59岁  17.2  1
36  16031284  男  52岁  21.1  1
37  16031822  女  58岁  19.3  1
38  16037970  男  46岁    30  1
39  16059736  女  53岁  15.3  1
40  16065569  男  54岁  17.8  1
41  16086039  女  57岁  18.9  1
42  16108185  女  29岁  16.8  1
43  16147283  女  60岁  25.1  1
44  16159443  男  60岁  12.7  1
45  16171595  女  52岁    17  1
46  16182203  男  54岁  16.6  1
47  16203411  女  57岁  14.7  1
48  16208480  女  55岁  21.5  1
49  16228130  女  59岁  16.5  1
50  16238261  男  49岁  24.6  1
51  16246609  女  60岁  26.4  1
52  16249435  女  55岁   8.5  1
53  16253143  男  52岁  22.9  1
54  16293130  女  59岁  28.7  1
55  16335597  男  60岁  18.8  1
56  16351149  女  56岁  22.2  1
57  16360047  女  57岁  15.6  1
58  16393326  女  51岁  18.1  1

In [85]:
for i in range(59):
    if df_test.loc[i,"性别"] == "男":
        df_test.loc[i,"性别"] = 1
    else:
        df_test.loc[i,"性别"] = 0

In [86]:
df_test

病人编号 性别   年龄  手测大小 性质
0   15948741  0  49岁    13  0
1   15953722  1  49岁  14.3  0
2   15969517  0  67岁    11  0
3   15973818  0  36岁   8.6  0
4   16002097  0  51岁  10.1  0
5   16013798  0  64岁  15.8  0
6   16015123  1  62岁  21.4  0
7   16016891  0  54岁  11.1  0
8   16037928  0  42岁  15.8  0
9   16037929  0  42岁  11.3  0
10  16061526  0  52岁   6.9  0
11  16115683  0  52岁  15.5  0
12  16125690  0  47岁  2.91  0
13  16138538  1  61岁  11.2  0
14  16146671  0  59岁  13.2  0
15  16179449  0  65岁  22.4  0
16  16190391  0  61岁  12.1  0
17  16205881  0  55岁   9.6  0
18  16246061  0  63岁  12.9  0
19  16249434  0  55岁  17.3  0
20  16292540  0  64岁  16.7  0
21  16293872  0  39岁  11.6  0
22  16327554  0  55岁   6.3  0
23  16332422  1  58岁  13.3  0
24  16341295  0  58岁   9.9  0
25  16344725  0  64岁  13.1  0
26  16356575  0  58岁  14.1  0
27  16372828  0  57岁  17.6  0
28  16380887  0  55岁   8.8  0
29  16388544  0  61岁  11.5  0
30  15957645  0  54岁    25  1
31  15959386  0  57岁  23.6  1
32  15975538  1  56岁  20.5  1
33  15987050  0  49岁  13.5  1
34  15999229  1  59岁  2.62  1
35  16030518  0  59岁  17.2  1
36  16031284  1  52岁  21.1  1
37  16031822  0  58岁  19.3  1
38  16037970  1  46岁    30  1
39  16059736  0  53岁  15.3  1
40  16065569  1  54岁  17.8  1
41  16086039  0  57岁  18.9  1
42  16108185  0  29岁  16.8  1
43  16147283  0  60岁  25.1  1
44  16159443  1  60岁  12.7  1
45  16171595  0  52岁    17  1
46  16182203  1  54岁  16.6  1
47  16203411  0  57岁  14.7  1
48  16208480  0  55岁  21.5  1
49  16228130  0  59岁  16.5  1
50  16238261  1  49岁  24.6  1
51  16246609  0  60岁  26.4  1
52  16249435  0  55岁   8.5  1
53  16253143  1  52岁  22.9  1
54  16293130  0  59岁  28.7  1
55  16335597  1  60岁  18.8  1
56  16351149  0  56岁  22.2  1
57  16360047  0  57岁  15.6  1
58  16393326  0  51岁  18.1  1

In [87]:
for i in range(59):
    age = df_test.loc[i,"年龄"]
    age = age[0:2]
    df_test.loc[i,"年龄"] = age

In [88]:
df_test

病人编号 性别  年龄  手测大小 性质
0   15948741  0  49    13  0
1   15953722  1  49  14.3  0
2   15969517  0  67    11  0
3   15973818  0  36   8.6  0
4   16002097  0  51  10.1  0
5   16013798  0  64  15.8  0
6   16015123  1  62  21.4  0
7   16016891  0  54  11.1  0
8   16037928  0  42  15.8  0
9   16037929  0  42  11.3  0
10  16061526  0  52   6.9  0
11  16115683  0  52  15.5  0
12  16125690  0  47  2.91  0
13  16138538  1  61  11.2  0
14  16146671  0  59  13.2  0
15  16179449  0  65  22.4  0
16  16190391  0  61  12.1  0
17  16205881  0  55   9.6  0
18  16246061  0  63  12.9  0
19  16249434  0  55  17.3  0
20  16292540  0  64  16.7  0
21  16293872  0  39  11.6  0
22  16327554  0  55   6.3  0
23  16332422  1  58  13.3  0
24  16341295  0  58   9.9  0
25  16344725  0  64  13.1  0
26  16356575  0  58  14.1  0
27  16372828  0  57  17.6  0
28  16380887  0  55   8.8  0
29  16388544  0  61  11.5  0
30  15957645  0  54    25  1
31  15959386  0  57  23.6  1
32  15975538  1  56  20.5  1
33  15987050  0  49  13.5  1
34  15999229  1  59  2.62  1
35  16030518  0  59  17.2  1
36  16031284  1  52  21.1  1
37  16031822  0  58  19.3  1
38  16037970  1  46    30  1
39  16059736  0  53  15.3  1
40  16065569  1  54  17.8  1
41  16086039  0  57  18.9  1
42  16108185  0  29  16.8  1
43  16147283  0  60  25.1  1
44  16159443  1  60  12.7  1
45  16171595  0  52    17  1
46  16182203  1  54  16.6  1
47  16203411  0  57  14.7  1
48  16208480  0  55  21.5  1
49  16228130  0  59  16.5  1
50  16238261  1  49  24.6  1
51  16246609  0  60  26.4  1
52  16249435  0  55   8.5  1
53  16253143  1  52  22.9  1
54  16293130  0  59  28.7  1
55  16335597  1  60  18.8  1
56  16351149  0  56  22.2  1
57  16360047  0  57  15.6  1
58  16393326  0  51  18.1  1

In [89]:
df_test.columns=["patient_num","sex","age","diameter","type"]  

In [90]:
df_test

patient_num sex age diameter type
0     15948741   0  49       13    0
1     15953722   1  49     14.3    0
2     15969517   0  67       11    0
3     15973818   0  36      8.6    0
4     16002097   0  51     10.1    0
5     16013798   0  64     15.8    0
6     16015123   1  62     21.4    0
7     16016891   0  54     11.1    0
8     16037928   0  42     15.8    0
9     16037929   0  42     11.3    0
10    16061526   0  52      6.9    0
11    16115683   0  52     15.5    0
12    16125690   0  47     2.91    0
13    16138538   1  61     11.2    0
14    16146671   0  59     13.2    0
15    16179449   0  65     22.4    0
16    16190391   0  61     12.1    0
17    16205881   0  55      9.6    0
18    16246061   0  63     12.9    0
19    16249434   0  55     17.3    0
20    16292540   0  64     16.7    0
21    16293872   0  39     11.6    0
22    16327554   0  55      6.3    0
23    16332422   1  58     13.3    0
24    16341295   0  58      9.9    0
25    16344725   0  64     13.1    0
26    16356575   0  58     14.1    0
27    16372828   0  57     17.6    0
28    16380887   0  55      8.8    0
29    16388544   0  61     11.5    0
30    15957645   0  54       25    1
31    15959386   0  57     23.6    1
32    15975538   1  56     20.5    1
33    15987050   0  49     13.5    1
34    15999229   1  59     2.62    1
35    16030518   0  59     17.2    1
36    16031284   1  52     21.1    1
37    16031822   0  58     19.3    1
38    16037970   1  46       30    1
39    16059736   0  53     15.3    1
40    16065569   1  54     17.8    1
41    16086039   0  57     18.9    1
42    16108185   0  29     16.8    1
43    16147283   0  60     25.1    1
44    16159443   1  60     12.7    1
45    16171595   0  52       17    1
46    16182203   1  54     16.6    1
47    16203411   0  57     14.7    1
48    16208480   0  55     21.5    1
49    16228130   0  59     16.5    1
50    16238261   1  49     24.6    1
51    16246609   0  60     26.4    1
52    16249435   0  55      8.5    1
53    16253143   1  52     22.9    1
54    16293130   0  59     28.7    1
55    16335597   1  60     18.8    1
56    16351149   0  56     22.2    1
57    16360047   0  57     15.6    1
58    16393326   0  51     18.1    1

In [91]:
test_group = [1]*59

In [92]:
df_test.insert(loc=0, column='group', value=test_group)

In [247]:
df_test

group patient_num sex age diameter type
0       1    15948741   0  49       13    0
1       1    15953722   1  49     14.3    0
2       1    15969517   0  67       11    0
3       1    15973818   0  36      8.6    0
4       1    16002097   0  51     10.1    0
5       1    16013798   0  64     15.8    0
6       1    16015123   1  62     21.4    0
7       1    16016891   0  54     11.1    0
8       1    16037928   0  42     15.8    0
9       1    16037929   0  42     11.3    0
10      1    16061526   0  52      6.9    0
11      1    16115683   0  52     15.5    0
12      1    16125690   0  47     2.91    0
13      1    16138538   1  61     11.2    0
14      1    16146671   0  59     13.2    0
15      1    16179449   0  65     22.4    0
16      1    16190391   0  61     12.1    0
17      1    16205881   0  55      9.6    0
18      1    16246061   0  63     12.9    0
19      1    16249434   0  55     17.3    0
20      1    16292540   0  64     16.7    0
21      1    16293872   0  39     11.6    0
22      1    16327554   0  55      6.3    0
23      1    16332422   1  58     13.3    0
24      1    16341295   0  58      9.9    0
25      1    16344725   0  64     13.1    0
26      1    16356575   0  58     14.1    0
27      1    16372828   0  57     17.6    0
28      1    16380887   0  55      8.8    0
29      1    16388544   0  61     11.5    0
30      1    15957645   0  54       25    1
31      1    15959386   0  57     23.6    1
32      1    15975538   1  56     20.5    1
33      1    15987050   0  49     13.5    1
34      1    15999229   1  59     2.62    1
35      1    16030518   0  59     17.2    1
36      1    16031284   1  52     21.1    1
37      1    16031822   0  58     19.3    1
38      1    16037970   1  46       30    1
39      1    16059736   0  53     15.3    1
40      1    16065569   1  54     17.8    1
41      1    16086039   0  57     18.9    1
42      1    16108185   0  29     16.8    1
43      1    16147283   0  60     25.1    1
44      1    16159443   1  60     12.7    1
45      1    16171595   0  52       17    1
46      1    16182203   1  54     16.6    1
47      1    16203411   0  57     14.7    1
48      1    16208480   0  55     21.5    1
49      1    16228130   0  59     16.5    1
50      1    16238261   1  49     24.6    1
51      1    16246609   0  60     26.4    1
52      1    16249435   0  55      8.5    1
53      1    16253143   1  52     22.9    1
54      1    16293130   0  59     28.7    1
55      1    16335597   1  60     18.8    1
56      1    16351149   0  56     22.2    1
57      1    16360047   0  57     15.6    1
58      1    16393326   0  51     18.1    1

In [93]:
pre_test = np.load("D:/python/pre_test.npy")

In [94]:
df_test.insert(loc=5, column='deepscore', value=pre_test)

In [95]:
df_test

group patient_num sex age diameter  deepscore type
0       1    15948741   0  49       13   0.245279    0
1       1    15953722   1  49     14.3   0.181513    0
2       1    15969517   0  67       11   0.683991    0
3       1    15973818   0  36      8.6   0.583086    0
4       1    16002097   0  51     10.1   0.292922    0
5       1    16013798   0  64     15.8   0.589811    0
6       1    16015123   1  62     21.4   0.908520    0
7       1    16016891   0  54     11.1   0.775610    0
8       1    16037928   0  42     15.8   0.014810    0
9       1    16037929   0  42     11.3   0.848919    0
10      1    16061526   0  52      6.9   0.543883    0
11      1    16115683   0  52     15.5   0.585344    0
12      1    16125690   0  47     2.91   0.459998    0
13      1    16138538   1  61     11.2   0.272163    0
14      1    16146671   0  59     13.2   0.584935    0
15      1    16179449   0  65     22.4   0.459310    0
16      1    16190391   0  61     12.1   0.276215    0
17      1    16205881   0  55      9.6   0.413527    0
18      1    16246061   0  63     12.9   0.413012    0
19      1    16249434   0  55     17.3   0.673382    0
20      1    16292540   0  64     16.7   0.134069    0
21      1    16293872   0  39     11.6   0.428470    0
22      1    16327554   0  55      6.3   0.712859    0
23      1    16332422   1  58     13.3   0.781334    0
24      1    16341295   0  58      9.9   0.465992    0
25      1    16344725   0  64     13.1   0.106252    0
26      1    16356575   0  58     14.1   0.705071    0
27      1    16372828   0  57     17.6   0.083716    0
28      1    16380887   0  55      8.8   0.526876    0
29      1    16388544   0  61     11.5   0.456657    0
30      1    15957645   0  54       25   0.268892    1
31      1    15959386   0  57     23.6   0.105052    1
32      1    15975538   1  56     20.5   0.310071    1
33      1    15987050   0  49     13.5   0.239793    1
34      1    15999229   1  59     2.62   0.144657    1
35      1    16030518   0  59     17.2   0.217644    1
36      1    16031284   1  52     21.1   0.415176    1
37      1    16031822   0  58     19.3   0.043181    1
38      1    16037970   1  46       30   0.116199    1
39      1    16059736   0  53     15.3   0.056358    1
40      1    16065569   1  54     17.8   0.483704    1
41      1    16086039   0  57     18.9   0.323851    1
42      1    16108185   0  29     16.8   0.186773    1
43      1    16147283   0  60     25.1   0.068162    1
44      1    16159443   1  60     12.7   0.488073    1
45      1    16171595   0  52       17   0.263950    1
46      1    16182203   1  54     16.6   0.192246    1
47      1    16203411   0  57     14.7   0.293343    1
48      1    16208480   0  55     21.5   0.147233    1
49      1    16228130   0  59     16.5   0.281326    1
50      1    16238261   1  49     24.6   0.369272    1
51      1    16246609   0  60     26.4   0.185731    1
52      1    16249435   0  55      8.5   0.488468    1
53      1    16253143   1  52     22.9   0.349770    1
54      1    16293130   0  59     28.7   0.164075    1
55      1    16335597   1  60     18.8   0.288111    1
56      1    16351149   0  56     22.2   0.110194    1
57      1    16360047   0  57     15.6   0.396045    1
58      1    16393326   0  51     18.1   0.839047    1

In [96]:
df_test.to_csv('D:/python/df_train.csv')

In [102]:
df_total = df_train.append(df_test, ignore_index=True)

In [104]:
df_total

group patient_num sex age diameter  deepscore type
0        0     7826653   1  52     11.1   0.965127    0
1        0    10554277   1  53     10.7   0.515213    0
2        0    11086905   0  56      8.7   0.876166    0
3        0    11086906   0  56      6.6   0.582331    0
4        0    11128066   1  64     15.2   0.791398    0
..     ...         ...  ..  ..      ...        ...  ...
205      1    16293130   0  59     28.7   0.164075    1
206      1    16335597   1  60     18.8   0.288111    1
207      1    16351149   0  56     22.2   0.110194    1
208      1    16360047   0  57     15.6   0.396045    1
209      1    16393326   0  51     18.1   0.839047    1

[210 rows x 7 columns]

for i in range(210):
    if df_total.loc[i,"type"] == 0:
        a = df_total.loc[i,"diameter"] 
        df_total.loc[i,"diameter"] = a + 1
    else:
         b = df_total.loc[i,"diameter"] 
         df_total.loc[i,"diameter"] = b - 1

In [105]:
df_total

group patient_num sex age diameter  deepscore type
0        0     7826653   1  52     11.1   0.965127    0
1        0    10554277   1  53     10.7   0.515213    0
2        0    11086905   0  56      8.7   0.876166    0
3        0    11086906   0  56      6.6   0.582331    0
4        0    11128066   1  64     15.2   0.791398    0
..     ...         ...  ..  ..      ...        ...  ...
205      1    16293130   0  59     28.7   0.164075    1
206      1    16335597   1  60     18.8   0.288111    1
207      1    16351149   0  56     22.2   0.110194    1
208      1    16360047   0  57     15.6   0.396045    1
209      1    16393326   0  51     18.1   0.839047    1

[210 rows x 7 columns]

In [106]:
df_total.to_csv('D:/python/df_total.csv', index=0)

# 基线资料统计学分析

In [214]:
# plotnine：python中的ggplot2
import numpy as np
import pandas as pd

# 统计分析
from scipy import stats
import statsmodels.api as sm
from statsmodels.formula.api import ols, glm, poisson

import copy

# 阻止pandas产生warnings（提示DataFrame的相关操作生成引用or副本）
import warnings
warnings.filterwarnings("ignore")

# jupyter中同一个cell的多个结果均自动输出，不需挨个手工print
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"


In [215]:
df_train["diameter"].mean()

16.152516556291385

In [216]:
# 标准误
se = np.std(df_train["diameter"]) / np.sqrt(len(df_train["diameter"]))
se

0.4925408455421459

In [217]:
# 区间上下限
1.96 * se

0.9653800572626059

In [218]:
stats.shapiro(df_train["diameter"])

ShapiroResult(statistic=0.9672616124153137, pvalue=0.0011594499228522182)